src/snowflake/connector/auth/_auth.py

```diff
@@ -307,6 +309,7 @@ def post_request_wrapper(self, url, headers, body) -> None:
                 while not self.ret or self.ret.get("message") == "Timeout":
                     next(c)
             else:
+                # no need to set a timeout on join as _post_request will terminate on timeout
```

Comment:
I would rather change the comment to something like "timeout safeguard", otherwise the comment seems to conflict the implementation of next line